### PR TITLE
Feat: add polar consistency test

### PR DESCRIPTION
**Pull request by anyangml**
This PR is to add cross framework consistency test on PolarFittingNet.

Note: `shift_diag` not yet implemented in PT.

### CODE REVIEW

File: source/tests/consistent/fitting/test_polar.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        (
+            resnet_dt,
+            precision,
+            mixed_types,
+        ) = self.param
+        if precision == "float64":
+            return 1e-10
+        elif precision == "float32":
+            return 1e-3
+        else:
```
Comment (anyangml) [inline source/tests/consistent/fitting/test_polar.py:176]: won't pass `1e-4`

File: deepmd/utils/argcheck.py [hunk 2]
```diff
@@ -718,9 +718,7 @@ def fitting_polar():
         Argument("resnet_dt", bool, optional=True, default=True, doc=doc_resnet_dt),
         Argument("precision", str, optional=True, default="default", doc=doc_precision),
         Argument("fit_diag", bool, optional=True, default=True, doc=doc_fit_diag),
-        Argument(
-            "scale", [List[float], float], optional=True, default=1.0, doc=doc_scale
-        ),
+        Argument("scale", List[float], optional=True, default=None, doc=doc_scale),
```
Comment (njzjz) [inline deepmd/utils/argcheck.py:721]: Is there any reason to change it? The argument is exposed to users. Changing it will cause a breaking change.

Comment (njzjz) [reply deepmd/utils/argcheck.py:721]: It's okay if `None` is equivalent to `1.0`, but it's a breaking change to disallow `float`.

Comment (anyangml) [reply deepmd/utils/argcheck.py:721]: I see, I will update torch to match this behavior.

**Review by njzjz — APPROVED** [review]

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**